src/app_service/service/community/service.nim

```diff
@@ -1351,6 +1346,60 @@ QtObject:
       self.events.emit(SIGNAL_COMMUNITY_IMPORTED, CommunityArgs(community: community))
 
     self.events.emit(SIGNAL_COMMUNITY_DATA_IMPORTED, CommunityArgs(community: community))
+  
+  proc removeLoadingRequestToJoinCommunity(self: Service, communityId: string) =
+    var i = 0
+    for communityRequest in self.myCommunityRequests:
+      if (communityRequest.communityId == communityId and RequestToJoinType(communityRequest.state) == RequestToJoinType.Loading):
+          self.myCommunityRequests.delete(i)
+          self.events.emit(SIGNAL_REQUEST_TO_JOIN_COMMUNITY_DELETED, CommunityRequestToJoinArgs(communityId: communityId))
+          return
+      i.inc()
+
+  proc asyncRequestToJoinCommunity*(self: Service, communityId: string, ensName: string, password: string) =
+    try:
+      let communityRequest = CommunityMembershipRequestDto(
+        id: "",
+        publicKey: "",
+        chatId: "",
+        communityId: communityId,
+        state: int(RequestToJoinType.Loading),
+        our: ""
+      )
+      self.myCommunityRequests.add(communityRequest)
+      self.events.emit(SIGNAL_COMMUNITY_MY_REQUEST_ADDED, CommunityRequestArgs(communityRequest: communityRequest))
+
+      let arg = AsyncRequestToJoinCommunityTaskArg(
+        tptr: cast[ByteAddress](asyncRequestToJoinCommunityTask),
+        vptr: cast[ByteAddress](self.vptr),
+        slot: "onAsyncRequestToJoinCommunityDone",
+        communityId: communityId,
+        ensName: ensName,
+        password: password
+      )
+      self.threadpool.start(arg)
+    except Exception as e:
+      self.removeLoadingRequestToJoinCommunity(communityId)
+      error "Error request to join community", msg = e.msg 
+    
+  proc onAsyncRequestToJoinCommunityDone*(self: Service, communityIdAndRpcResponse: string) {.slot.} =
+    try:
+      let rpcResponseObj = communityIdAndRpcResponse.parseJson
+      if (rpcResponseObj{"response"}{"error"}.kind != JNull):
+        let error = Json.decode($rpcResponseObj["response"]["error"], RpcError)
+        error "Error requesting community info", msg = error.message
+        return
+
+      let communityId = rpcResponseObj{"communityId"}.getStr()
+      let rpcResponse = Json.decode($rpcResponseObj["response"], RpcResponse[JsonNode])
+      self.activityCenterService.parseActivityCenterResponse(rpcResponse)
+      
+      self.removeLoadingRequestToJoinCommunity(communityId)
```

Comment:
Do we really want to remove the request to join when we get the answer? I think it most cases, this will go in "pending" state, so we don't want to delete it, just change the state?

Maybe we just need to change the event from `DELETED` to `UPDATED` so that it can work for all the state changes `loading -> pending -> accepted/rejected`

Reply:
I thought about it, but request there is kind a `fake` cos in database it's not exists. Technically you may be right, will fix it.

Reply:
Updated :)

Reply:
Delete method still make sense as a clean up method